lib/super_digit.rb

```diff
@@ -1,15 +1,33 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) where n is the number of digits in the input number
+# Space Complexity - O(log n)
 def super_digit(n)
-   
+  puts "super_digit("+n.to_s+")"
+  if (n / 10) == 0
+    return n
+  else
+    sum = 0
+    while ((n / 10.0) > 0)
+      puts "  while"
+      sum += (n % 10)
+      n = (n / 10)
+    end
+    return super_digit(sum)
+  end
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+#     because the while loop to create the concatenated number
+#     and the while loop adding the digits of the number are not nested.
+# Space Complexity - O(log k * n)
 def refined_super_digit(n, k)
```

Comment:
This is the brute-force solution here.  Can you think of a better way to do this with dynamic programming?